### PR TITLE
Rework converter interface + added faros destination write test

### PR DESCRIPTION
**Pull request by tovbinm**
## Description

Rework converter interface + added faros destination write test

## Type of change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

### CODE REVIEW

File: destinations/faros-destination/src/converters/converter-registry.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        }
+        ConverterRegistry.convertersByStream[name] = converter;
+      }
+      ConverterRegistry.initialized = true;
+    }
+    return ConverterRegistry.convertersByStream[streamName];
+  }
+
+  private static streamNameToString(name: StreamName): string {
+    return `${name.prefix}__${name.name}`;
```
Comment (cjwooo) [inline destinations/faros-destination/src/converters/converter-registry.ts:39]: I thought the format was gonna be `<sourceName>__<sourceType>__<streamName>`

File: destinations/faros-destination/test/temp.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import fs from 'fs';
+import {AffixOptions, open, track} from 'temp';
+
+// TODO: this does not seem to work (ask Chris)
+// Automatically track and cleanup temp files at exit
+track();
```
Comment (cjwooo) [inline destinations/faros-destination/test/temp.ts:6]: Maybe try calling track at the start of the tempFile function

File: destinations/faros-destination/src/converters/converter-registry.ts (hunk 1 above)
Comment (tovbinm) [reply destinations/faros-destination/src/converters/converter-registry.ts]: fixed

Comment (tovbinm) [reply destinations/faros-destination/src/converters/converter-registry.ts]: I am actually ignoring anything before source type, e.g `<...>__<...ignored....>__<source>__<name>`.

**Comment by tovbinm** [comment]
@cjwooo can you check if the build & tests pass on your machine?

**Review by cjwooo — APPROVED** [review]

**Pull request merged**